src/metpy/plots/declarative.py

```diff
@@ -1040,13 +1086,14 @@ def griddata(self):
     def plotdata(self):
         """Return the data for plotting.
 
-        The data array, x coordinates, and y coordinates.
+        The two dimension coordinates and the data array.
 
         """
-        x = self.griddata.metpy.x
-        y = self.griddata.metpy.y
-
-        return x, y, self.griddata
+        return (
+            self.griddata[self.griddata.dims[1]],
```

Comment:
Doesn't this take us back to an era where we depended up on the order of the dimensions? Couldn't we pick one or the other of `x` and `longitude`?

Reply:
Good catch, I missed that in https://github.com/jthielen/MetPy/commit/cfef550ea81b35213cab84ea6a1f44bca93e1b53. Though I do wonder based on not seeing any transpose/shape handling if xy-ordered data worked before this change? (Or I could just be totally missing something...I haven't worked with xy-ordered data in quite some time.)

However, we definitely can't just pick the physical `x` dimension coordinate, since this is an x axis only in the plotting sense, data-wise it could be any of MetPy's axis types. Given that there are six possible dimension pairs (x-y, x-vertical, x-time, y-vertical, y-time, and vertical-time), should we force/assume which of the pair goes on which axis or find a way to leave it up to the user? Or find some compromise in between?

Reply:
So in the end, we will always have two dimensional data, and we now have better checks for that with a warning. We could default to the ordering of the data based on our most common current use case (y-x) and build in some sensible choices based on the names of dimensions, but otherwise include an attribute like `swap_axes` that could be boolean (which defaults to `False`) to allow a change on the fly.